scheduler/scheduler.go

```diff
@@ -185,7 +185,7 @@ func (s *Scheduler) Sync(ctx context.Context, client schema.ClientMeta, tables s
 	// send migrate messages first
 	for _, tableOriginal := range tables.FlattenTables() {
 		migrateMessage := &message.SyncMigrateTable{
-			Table: tableOriginal.Copy(nil),
+			Table: tableOriginal.Copy(tableOriginal.Parent),
```

Comment:
note that we need to be careful around the `schema.Tables.FlattenTables` call around such change

Reply:
Please note we don't mutate `tableOriginal` or the parent, and the table copy will be disregarded once the migrate table message is processed